mixins/loading-complete/loading-complete-mixin.js

```diff
@@ -1,21 +1,36 @@
-export const LoadingCompleteMixin = (superclass) => class extends superclass {
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
 
-	constructor() {
-		super();
-		this._loadingCompletePromise = new Promise(resolve => {
-			this._loadingCompleteResolve = resolve;
-		});
+export const LoadingCompleteMixin = dedupeMixin((superclass) => class extends superclass {
+
+	#loadingCompleteResolve;
+
+	// eslint-disable-next-line sort-class-members/sort-class-members
+	#loadingCompletePromise = this.getLoadingComplete === this.#getLoadingComplete
+		? new Promise(resolve => this.#loadingCompleteResolve = resolve)
+		: Promise.resolve();
+
+	get getLoadingComplete() {
```

Comment:
Haha I think I'm getting confused by all the different things called `getLoadingComplete`. For my own sanity, is this correct?
- This is a getter for the `getLoadingComplete()` method that consumers can optionally override to implement custom logic. If they do override it, they're still expected to wait for `super.getLoadingComplete()`.
- If they don't override it, the default behaviour is to return a reference to the _private_ `#getLoadingComplete()` method
- That private method waits for `super.getLoadingComplete` in case other mixins/etc. have things to wait for and then finally waits for the private `#loadingComplete` Promise

What's the purpose of the private method in all this? Couldn't its code just move into here?

Reply:
I got rid of the private method by using `hasOwn` instead of comparing the two methods for equality, which didn't work if the mixin chain was swapped around. 2-for-1 there.